tests/phpunit/tests/term/isTermPubliclyViewable.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * @group term
+ */
+class Tests_Term_IsTermPubliclyViewable extends WP_UnitTestCase {
+	/**
+	 * Unit tests for is_term_publicly_viewable().
+	 *
+	 * @dataProvider data_is_term_publicly_viewable
+	 * @ticket 56215
+	 *
+	 * @param string $taxonomy The taxonomy name.
+	 * @param bool   $expected The expected result of the function call.
+	 */
+	public function test_is_term_publicly_viewable( $taxonomy, $expected, $parent_key = '' ) {
+		$term_id = $this->factory()->term->create(
+			array(
+				'taxonomy' => $taxonomy,
+			)
+		);
+
+		$this->assertSame( $expected, is_term_publicly_viewable( $term_id ) );
+	}
+
+	/**
+	 * Unit tests for is_term_publicly_viewable().
+	 *
+	 * @ticket 56215
+	 */
+	public function test_non_existent_term_is_not_publicly_viewable() {
+		$this->assertFalse( is_term_publicly_viewable( 123 ) );
+	}
+
```

Comment:
Are you able to move this above `test_is_term_publicly_viewable()`? I think convention is to keep the test and data methods consecutive whenever possible.